File: tools/pokemon/mon_icon.py
import cv2
import os
import shutil

if os.path.exists("result"):
    shutil.rmtree("result")
os.mkdir("result")
# https://www.spriters-resource.com/fullview/64342/
img = cv2.imread("mon_icon.png", -1)

names = [
    "mon",
    "bird",
    "water",
    "fairy",
    "grass",
    "bug",
    "snake",
    "quadruped",
    "ball",
    "helix",
]

for i in range(10):
    name = names[i]
    if i < 8:
        for j in range(2):
            x0 = j*16
            y0 = i*16
            x1 = x0 + 16
            y1 = y0 + 16

            tile = img[y0:y1, x0:x1]
            cv2.imwrite("./result/{}_mon_icon_{}.png".format(name, j), tile)
    else:
        x0 = 0
        y0 = i*16
        x1 = x0 + 16
        y1 = y0 + 16

        tile = img[y0:y1, x0:x1]
        cv2.imwrite("./result/{}_mon_icon_{}.png".format(name, 0), tile)

        x0 = 0
        y0 = i*16-1
        x1 = x0 + 16
        y1 = y0 + 16

        tile = img[y0:y1, x0:x1]
        cv2.imwrite("./result/{}_mon_icon_{}.png".format(name, 1), tile)

File: pkg/pkmn/monicon.go
package pkmn

import (
	"pokered/pkg/data/pkmnd"
	"pokered/pkg/store"
	"pokered/pkg/util"

	ebiten "github.com/hajimehoshi/ebiten/v2"
)

const (
	suffix    = "_mon_icon"
	ball      = "ball"
	bird      = "bird"
	bug       = "bug"
	fairy     = "fairy"
	grass     = "grass"
	helix     = "helix"
	mon       = "mon"
	quadruped = "quadruped"
	snake     = "snake"
	water     = "water"
)

var IconGen1 = map[uint][2]*ebiten.Image{
	pkmnd.BallMon: {
		util.OpenImage(store.FS, "/"+ball+suffix+"_0.png"),
		util.OpenImage(store.FS, "/"+ball+suffix+"_1.png"),
	},
	pkmnd.BirdMon: {
		util.OpenImage(store.FS, "/"+bird+suffix+"_0.png"),
		util.OpenImage(store.FS, "/"+bird+suffix+"_1.png"),
	},
	pkmnd.BugMon: {
		util.OpenImage(store.FS, "/"+bug+suffix+"_0.png"),
		util.OpenImage(store.FS, "/"+bug+suffix+"_1.png"),
	},
	pkmnd.FairyMon: {
		util.OpenImage(store.FS, "/"+fairy+suffix+"_0.png"),
		util.OpenImage(store.FS, "/"+fairy+suffix+"_1.png"),
	},
	pkmnd.GrassMon: {
		util.OpenImage(store.FS, "/"+grass+suffix+"_0.png"),
		util.OpenImage(store.FS, "/"+grass+suffix+"_1.png"),
	},
	pkmnd.HelixMon: {
		util.OpenImage(store.FS, "/"+helix+suffix+"_0.png"),
		util.OpenImage(store.FS, "/"+helix+suffix+"_1.png"),
	},
	pkmnd.MonMon: {
		util.OpenImage(store.FS, "/"+mon+suffix+"_0.png"),
		util.OpenImage(store.FS, "/"+mon+suffix+"_1.png"),
	},
	pkmnd.QuadrupedMon: {
		util.OpenImage(store.FS, "/"+quadruped+suffix+"_0.png"),
		util.OpenImage(store.FS, "/"+quadruped+suffix+"_1.png"),
	},
	pkmnd.SnakeMon: {
		util.OpenImage(store.FS, "/"+snake+suffix+"_0.png"),
		util.OpenImage(store.FS, "/"+snake+suffix+"_1.png"),
	},
	pkmnd.WaterMon: {
		util.OpenImage(store.FS, "/"+water+suffix+"_0.png"),
		util.OpenImage(store.FS, "/"+water+suffix+"_1.png"),
	},
}
